acr-stamp.bicep

```diff
@@ -0,0 +1,215 @@
+param geoRedundancyLocation string = 'centralus'
+
+/*** VARIABLES ***/
+
+var subRgUniqueString = uniqueString('aks', subscription().subscriptionId, resourceGroup().id)
+
+/*** EXISTING RESOURCES ***/
+
+resource spokeResourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' existing = {
+  scope: subscription()
+  name: '${split(targetVnetResourceId,'/')[4]}'
+}
+
+resource spokeVirtualNetwork 'Microsoft.Network/virtualNetworks@2021-05-01' existing = {
+  scope: spokeResourceGroup
+  name: '${last(split(targetVnetResourceId,'/'))}'
+  
+  resource snetClusterNodes 'subnets@2021-05-01' existing = {
+    name: 'snet-clusternodes'
+  }
+}
+
+/*** RESOURCES ***/
+
+resource laAks 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
+  name: 'la-aks-${subRgUniqueString}'
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+    retentionInDays: 30
+  }
+}
+
+// Limit access to a the ACR by assigning virtual network private IP addresses to the registry endpoints and using Azure Private Link
+resource dnsPrivateZoneAcr 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: 'privatelink.azurecr.io'
+  location: 'global'
+  properties: {}
+
+  resource dnsVnetLinkAcrToSpoke 'virtualNetworkLinks@2020-06-01' = {
+    name: 'to_${spokeVirtualNetwork.name}'
+    location: 'global'
+    properties: {
+      virtualNetwork: {
+        id: targetVnetResourceId
+      }
+      registrationEnabled: false
+    }
+  }
+}
+
+// The Container Registry that will be used when bootstrapped the AKS cluster
+resource acrAks 'Microsoft.ContainerRegistry/registries@2021-09-01' = {
+  name: 'acraks${subRgUniqueString}'
+  location: location
+  sku: {
+    name: 'Premium'
+  }
+  properties: {
+    adminUserEnabled: false
+    networkRuleSet: {
+      defaultAction: 'Deny'
+      ipRules: []
+    }
+    policies: {
+      quarantinePolicy: {
+        status: 'disabled'
+      }
+      trustPolicy: {
+        type: 'Notary'
+        status: 'disabled'
+      }
+      retentionPolicy: {
+        days: 15
+        status: 'enabled'
+      }
+    }
+    publicNetworkAccess: 'Disabled'
+    encryption: {
+      status: 'disabled'
+    }
+    dataEndpointEnabled: true
+    networkRuleBypassOptions: 'AzureServices'
+    zoneRedundancy: 'Disabled'
+  }
+
+  resource acrReplication 'replications@2021-09-01' = {
+    name: geoRedundancyLocation
+    location: geoRedundancyLocation
+    properties: {}
+  }
+}
+
+resource acrAks_diagnosticsSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  name: 'default'
+  scope: acrAks
+  properties: {
+    workspaceId: laAks.id
+    metrics: [
+      {
+        timeGrain: 'PT1M'
+        category: 'AllMetrics'
+        enabled: true
+      }
+    ]
+    logs: [
+      {
+        category: 'ContainerRegistryRepositoryEvents'
+        enabled: true
+      }
+      {
+        category: 'ContainerRegistryLoginEvents'
+        enabled: true
+      }
```

Comment:
Can you convert this to the `allLogs` category group?

Reply:
Done

Reply:
Thanks.